valhalla/midgard/gridded_data.h

```diff
@@ -181,8 +203,10 @@ template <std::size_t dimensions_t> class GriddedData : public Tiles<PointLL> {
     contours_t contours(intervals.size(), std::list<feature_t>{feature_t{}});
 
     // and something to find them quickly
-    using contour_lookup_t = std::unordered_map<PointLL, typename feature_t::iterator>;
-    std::vector<contour_lookup_t> lookups(intervals.size());
+    using contour_lookup_t = std::map<PointLL, typename feature_t::iterator>;
+    // store begins and ends of the segments separately not to loose segment orientation
+    std::vector<contour_lookup_t> begin_lookups(intervals.size());
+    std::vector<contour_lookup_t> end_lookups(intervals.size());
```

Comment:
we should quantify the performance loss due to these changes. i suppose we hope that its mostly outweighed by the actual expansion but in practice, i semi-recently refactored a bunch of this stuff to win back 10% performance in this area of the code (using the lambdas above)

Reply:
> I suppose we hope that its mostly outweighed by the actual expansion

I did some performance testing recently. Indeed, it's different orders of magnitude(i.e expansion takes > 90% comparing to non-expansion steps)

Reply:
can you post the results here. its important to know the changes in performance characteristics before we merge such changes imho

Reply:
I tested performance with `costing=auto&contour=60min` requests. I can barely see the difference in performance: it's close to 1-2% so looks mergeable to me

Reply:
Excellent, that sounds reasonable, thanks!